### ISSUE TITLE
Support for treating word families together.

### ISSUE THREAD

**Issue by simjanos-dev** [simjanos-dev/LinguaCafe#43]
Support for saving one translation for multiple words together. For example: 

å løpe/løper/løpt - to run.

**Comment by Einenlum** [comment]
Indeed, as you said in #96, it can also add complexity.

Let's take this example:

![aus](https://github.com/simjanos-dev/LinguaCafe/assets/5675200/d5cf9e5e-cfa6-454f-9209-ead9df757e21)

Here the lemma is "ausfahren", but the verb is split into two parts (which have different meanings when apart in a different context).
I know "aus" and I know "ausfahren". But if I set both to "this is known", then I will never learn and train about ausfahren. Not sure how to handle this, but I would prefer to learn about lemmas than specific words, at least in my case, when learning German.

**Comment by simjanos-dev** [comment]
How frequent is it? If it's not very common, you could modify the lemma to "ausfahren (1)" before saving the word, so they will be treated as a separate word. Currently words are being saved when you close the vocabulary box. 

If it is common and  will be important, maybe I could add a checkbox or something for every word, so you can uncheck them to treat the word as separate from their lemma.

**Comment by Einenlum** [comment]
In German it's unfortunately very common. For example, "hängen" means " to hang", "ab" means "from", but "abhängen" means "to depend on".

"Es **hängt** von der Situation **ab**" means "it depends on the situation".

Linguacafe already correctly detects "abhängen" as the lemma when I'm on "hängt" which is great, but since it applies the level and progression to words and not lemmas, it seems like I can't have a real training on "abhängen" unless it's written somewhere in the infinitive form. Not sure if my explanation is clear enough 🙂

**Comment by Einenlum** [comment]
And by the way, it's a very common issue with language learning apps when it comes to German. You can't apply a simple rule like "one word = one meaning", which makes it very complicated. It seems to me that Linguacafe detecting correctly the lemma (and making the distinction between lemmas and words) already made 80% of the hard work. I was positively surprised to see the right verb form detected.

**Comment by simjanos-dev** [comment]
Yeah, I understand it. When I learn Japanese, I usually treat every word as a different word, but there are not many versions of them, except for verbs. I can see how this is annoying. Also I don't use the SRS review system, I kind of review them randomly.

> Linguacafe already correctly detects "abhängen" as the lemma when I'm on "hängt" which is great, but since it applies the level and progression to words and not lemmas, it seems like I can't have a real training on "abhängen" unless it's written somewhere in the infinitive form

It depends on what you mean by training. When you review a word, you will also see its lemma. But if you want to see it in sentence, you will have to find an "abhängen" version in it.

I just realized that reviewing will also be problem if we group word levels. What if 2 different words with the same lemma has a shared level, and you review one of them? It should level up, but then what should happen to the other word? I think you should review both of them separately. But then it would level up twice. The first solution that comes to mind is you have to review them both, and only then they level up. It does complicate it.

Is levels being locked together important? If we only lock translations together, this would not cause a problem.

The credit for German verb support goes to @mbrine555. Huge thank you for it again!

**Comment by jzohrab** [comment]
Don't know if this helps, but Lute uses "parent terms" for root forms, and has a checkmark "link status" set at the child term level.  Terms can have multiple parents, because sometimes a word form is an  inflected verb, or a noun, or whatever.  Originally Lute only had independent statuses for all forms, but some people found that tiresome.

**Comment by simjanos-dev** [comment]
> and has a checkmark "link status" set at the child term level.

Probably that's what I'll do as well, so people can opt out of single words being connected to their other forms. I mean on top of adding a setting to treat every single word separate.

> Terms can have multiple parents, because sometimes a word form is an inflected verb, or a noun, or whatever.
I think at first step I will ignore those differences. 

Does managing multiple parents feel less tiresome? Did lute also had a single parent in the past? How did people find that change?

**Comment by jzohrab** [comment]
> so people can opt out of single words being connected to their other forms

Yes, that's necessary, some forms are just harder than others.  e.g. some special verb tenses/moods are nice to track differently. (Note: sometimes I feel that too many data options for users becomes a make-work feature, where they start thinking about _data_ instead of just _reading_ :-) , but I suppose it's useful in some cases.)

> managing multiple parents

Lute had single parents in the past but I changed it to multiple b/c people needed it in certain situations.  I use tagify to specify the parents.  Lute doesn't lemmatize at all, users have to do that, though there are offline/clumsy methods to load them :-).  No pushback on the change at all, it's likely only used in a few cases.

If a term has multiple parents, the statuses can't be linked.  I think that's ok, if it's handled differently then the load on the user gets silly.  There are notes in the [manual](https://jzohrab.github.io/lute-manual/usage/terms/link-parent-child-statuses.html) that might be helpful for your design ... maybe. :-)

ps - hope you don't mind me chiming in, I just happened to have worked through some of this already.  Not trying to pull focus from this project to Lute, LinguaCafe looks awesome. 👍

**Comment by Einenlum** [comment]
> It depends on what you mean by training. When you review a word, you will also see its lemma. But if you want to see it in sentence, you will have to find an "abhängen" version in it.

Yeah, I mean that it would be nice to say once for all "I know all the versions of abhängen, don't show me them as new terms in my books".

> I just realized that reviewing will also be problem if we group word levels. What if 2 different words with the same lemma has a shared level, and you review one of them? It should level up, but then what should happen to the other word? I think you should review both of them separately. But then it would level up twice. The first solution that comes to mind is you have to review them both, and only then they level up. It does complicate it.
> 
> Is levels being locked together important? If we only lock translations together, this would not cause a problem.

I think it could actually also be a config for the whole language: do you want to use lemmas or terms? I think some people (like me) don't really care about terms and only want to learn and spot lemmas. If you decided to use lemmas instead of words, then there could be an optional way to track a specific word instead of its lemma.
